erpnext/stock/doctype/purchase_receipt/purchase_receipt.py

```diff
@@ -118,6 +118,9 @@ def validate_with_previous_doc(self):
 
 	def po_required(self):
 		if frappe.db.get_value("Buying Settings", None, "po_required") == 'Yes':
+			if frappe.get_value('Supplier', self.supplier, 'allow_purchase_invoice_creation_without_purchase_order'):
```

Comment:
the checkbox clearly states that it affects Purchase Invoice, so, we only check this on Purchase Invoice or Sales Invoice validation. This just adds more unexplained behaviour which is unnecessary. Let's make the checkboxes function as per their labels.